node/subsystem-types/src/runtime_client.rs

```diff
@@ -252,7 +252,18 @@ where
 		&self,
 		at: Hash,
 	) -> Result<Vec<CoreState<Hash, BlockNumber>>, ApiError> {
-		self.runtime_api().availability_cores(at)
+		let version = self
```

Comment:
We discussed somewhere that such transformations in the client are not encouraged and we prefer doing this in the subsystem. Unfortunately I don't remember the exact reasons.

For your particular case I think it's okay though. Depending on the runtime api you call either the new or the old version of the function. And you do this workaround to avoid the `changed_in` issues with the runtime api.

Reply:
Which file would this ideally go to then? What exactly is the subsystem again? I'm only aware of the node/client and runtime distinction.

Reply:
Subsystem is a module (lets say) from the node - e.g. `approval-voting`, `dispute-coordinator` etc.
You can put a function in any helpers module (e.g. https://github.com/paritytech/polkadot/blob/dbae30efe080a1d41fe54ef4da8af47614c9ca93/node/subsystem-util/src/runtime/mod.rs) and call it each time you need the availability_cores.

But I'd suggest not to touch it for now.

Reply:
Subsystem is a module (lets say) from the node - e.g. `approval-voting`, `dispute-coordinator` etc.
You can put a function in any helpers module (e.g. https://github.com/paritytech/polkadot/blob/dbae30efe080a1d41fe54ef4da8af47614c9ca93/node/subsystem-util/src/runtime/mod.rs) and call it each time you need the availability_cores.

But I'd suggest not to touch it for now.